Add docker builder option to provider

aacebedo opened the pull request:
This PR solves #984 by first checking if podman is used when buildxExists is called and return true in this case.
It also adds a DOCKER_BUILDER option to the provider in order to choose easily between the 2 builders.

aacebedo commented:
@pascalbreuninger can you have a look to this one and give me some feedback please?

pascalbreuninger commented on cmd/build.go:
```diff
@@ -115,9 +115,7 @@ func NewBuildCmd(flags *flags.GlobalFlags) *cobra.Command {
 
 	// TESTING
 	buildCmd.Flags().BoolVar(&cmd.ForceBuild, "force-build", false, "TESTING ONLY")
```
nevermind, understood why but can we still keep the flag?

aacebedo replied:
I can re-add the flag if you prefer but I did that because the flag was hidden and it would lead to have two ways to do the same thing (force to use buildkit). I tend to follow KISS reason why I removed it.

aacebedo replied:
fixed

pascalbreuninger replied:
🙇 thanks - it's technically not necessary but I'd rather not break existing scripts

pascalbreuninger reviewed (changes requested):
@aacebedo thanks for the PR! Got a couple of comments and want to make sure tests are still working

aacebedo commented:
> @aacebedo thanks for the PR! Got a couple of comments and want to make sure tests are still working

Thanks for the review! Yes I will solve the e2e tests.

aacebedo commented on pkg/docker/helper.go:
```diff
@@ -17,6 +18,33 @@ import (
+
+// Enum values for DockerBuilder.
+const (
+	DockerBuilderBuildKit DockerBuilder = iota
```
Agreed

aacebedo replied:
fixed

aacebedo commented:
> > @aacebedo thanks for the PR! Got a couple of comments and want to make sure tests are still working
> 
> Thanks for the review! Yes I will solve the e2e tests.

@pascalbreuninger I have fixed the test. I have added a workflow dispatch event to E2E test to ease the next PRs. Don't know if it is ok for you but it helped me a lot as I do not have the same environment as the CI.

pascalbreuninger commented:
> @pascalbreuninger I have fixed the test. I have added a workflow dispatch event to E2E test to ease the next PRs. Don't know if it is ok for you but it helped me a lot as I do not have the same environment as the CI.

One small cleanup remaining and we're good to merge. 
The workflow dispatch makes sense 👍 Also just doubled checked and apparently only collaborators and above can trigger it in the main repo. Just wanted to be sure we're not opening an avenue for burning through our Github Action minutes 😅

aacebedo commented on pkg/docker/helper.go:
```diff
@@ -17,6 +18,33 @@ import (
+)
+
+func (db DockerBuilder) String() string {
+	return [...]string{"buildkit", "buildx", ""}[db]
```
corrected

pascalbreuninger reviewed (approved):
LGTM, thanks 🙌
